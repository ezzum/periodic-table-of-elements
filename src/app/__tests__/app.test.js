import { mount } from "enzyme";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";

import App from "../app";
import React from "react";

jest.mock(".././pages/table/table", () => "div");
jest.mock(".././pages/description/description", () => "div");
jest.mock(".././error-boundary/error-boundary", () => "div");

describe("App tests", () => {
  const fakeElements = [
    {
      symbol: "H",
      number: 1,
      name: "водород",
      atomicMass: 1.007,
      column: 1,
      row: 1,
      group: "nonMetal",
    },
  ];
  const initialStore = {};
  const mockStore = configureStore();
  let store, wrapper;

  beforeEach(() => {
    jest.spyOn(global, "fetch").mockImplementation(() => {
      Promise.resolve({
        json: () => Promise.resolve(fakeElements),
      });
    });
    store = mockStore(initialStore);
    wrapper = async () => {
      mount(
        <Provider store={store}>
          <App />
        </Provider>
      );
    };
  });

  it("snapshot test", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
