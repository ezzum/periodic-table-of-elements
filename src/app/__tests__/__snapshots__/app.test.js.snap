// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests snapshot test 1`] = `[Function]`;
